src/FamilyHubs.Referral.Core/Commands/CreateReferral/CreateReferralCommand.cs

```diff
@@ -148,24 +149,32 @@ private async Task<Data.Entities.Referral> AttachExistingService(Data.Entities.R
                 throw new ArgumentException($"Failed to return Service from service directory for Id = {entity.ReferralService.Id}");
             }
 
-            ServiceDirectory.Shared.Dto.OrganisationDto? sdOrganisation = await _serviceDirectoryService.GetOrganisationById(sdService.OrganisationId);
-            if (sdOrganisation == null)
+            // check if the organization already exists
+            //todo: do we need to update the organisation from the sd, if it already exists?
+            Organisation? organisation = await _context.Organisations.FindAsync(sdService.OrganisationId);
+            if (organisation == null)
             {
-                throw new ArgumentException($"Failed to return Organisation from service directory for Id = {sdService.OrganisationId}");
+                ServiceDirectory.Shared.Dto.OrganisationDto? sdOrganisation = await _serviceDirectoryService.GetOrganisationById(sdService.OrganisationId);
+                if (sdOrganisation == null)
+                {
+                    throw new ArgumentException($"Failed to return Organisation from service directory for Id = {sdService.OrganisationId}");
```

Comment:
nitpick: is this more of an `InvalidOperationExeption` rather than an `ArgumentException`, since the issue isn't that an argument for this method is missing, but the SD isn't returning what we expect?

Reply:
Yes I agree - I noticed that too, but I thought that changing the exception just before a release was too risky as consumer code might depend on the type of exception thrown.
There are other small issues in the code too, but I wanted the changes to be minimal at this point.